chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -357,6 +360,77 @@ case class ChainHandler(
   override def getFiltersAtHeight(height: Int)(
       implicit ec: ExecutionContext): Future[Vector[CompactFilterDb]] =
     filterDAO.getAtHeight(height)
+
+  /** @inheritdoc */
+  def getMatchingBlocks(
+      addresses: Vector[BitcoinAddress],
+      start: Option[BlockStamp] = None,
+      end: Option[BlockStamp] = None)(
+      implicit ec: ExecutionContext): Future[Vector[DoubleSha256DigestBE]] = {
+
+    logger.info(
+      s"Starting looking for matching blocks for addresses ${addresses.mkString(",")}")
+
+    val batchSize = chainConfig.filterBatchSize
+
+    val bytes = addresses.map(_.scriptPubKey.asmBytes)
+
+    val startHeightF =
+      Future.successful(start.map(BlockStamp.height).getOrElse(0))
+    val endHeightF = for {
+      filterCount <- getFilterCount
+    } yield end.map(BlockStamp.height).getOrElse(filterCount)
+
+    @tailrec
+    def scanFilters(
+        i: Int,
+        start: Int,
+        acc: Future[Vector[DoubleSha256DigestBE]]): Future[
+      Vector[DoubleSha256DigestBE]] = {
+      if (i <= start) {
+        acc.foreach { blocks =>
+          logger.info(s"Done looking for matching blocks for addresses ${addresses
+            .mkString(",")}: blocks matched ${blocks.size} latest block ${blocks.headOption
+            .getOrElse("")}")
+        }
+        acc.failed.foreach { e =>
+          logger.error(
+            s"Cannot find matching blocks for addresses ${addresses.mkString(",")}",
+            e)
+        }
+        acc
+      } else {
+        val newAcc: Future[Vector[DoubleSha256DigestBE]] = for {
+          compactFilterDbs <- filterDAO.getBetweenHeights(i - (batchSize - 1),
+                                                          i)
+          filtered <- Future.sequence(compactFilterDbs.map { filter =>
+            Future {
```

Comment:
Why do we need to wrap in another `Future` here?

Reply:
To let the execution context to run `filter.golombFilter.matchesAny(bytes)` in separate threads.